connectivity/check/test.go

```diff
@@ -189,6 +198,30 @@ func (t *Test) WithPolicy(policy string) *Test {
 		t.Fatalf("Parsing policy YAML: %s", err)
 	}
 
+	// Change the default test namespace as required.
+	for i := range pl {
+		pl[i].Namespace = t.ctx.params.TestNamespace
+		if pl[i].Spec != nil {
+			// Check both 'io.kubernetes.pod.namespace' and 'k8s:io.kubernetes.pod.namespace'.
+			for _, k := range []string{k8sConst.PodNamespaceLabel, kubernetesSourcedLabelPrefix + k8sConst.PodNamespaceLabel} {
+				for _, e := range pl[i].Spec.Egress {
+					for _, es := range e.ToEndpoints {
+						if n, ok := es.MatchLabels[k]; ok && n == defaults.ConnectivityCheckNamespace {
+							es.MatchLabels[k] = t.ctx.params.TestNamespace
+						}
+					}
+				}
+				for _, e := range pl[i].Spec.Ingress {
+					for _, es := range e.FromEndpoints {
+						if n, ok := es.MatchLabels[k]; ok && n == defaults.ConnectivityCheckNamespace {
+							es.MatchLabels[k] = t.ctx.params.TestNamespace
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
Just at a glance, this doesn't seem to be the exhaustive list of resources that might need a namespace change or is that not the case?

Reply:
Yes, this just affects the CNPs, but those are the only ones that need the namespace change (at least for now) since they're the only ones sourced from YAML files. Everything else (deployments, etc...) is built in Go code AFAIK.